lib/arjdbc/postgresql/adapter.rb

```diff
@@ -602,7 +511,11 @@ def query(sql, name = nil)
       log(sql, name) do
         result = []
         @connection.execute_query_raw(sql, nil) do |*values|
-          result << values
+          # We need to use #deep_dup here because it appears that
```

Comment:
I wasn't sure the best way to fix this. The issue was that the result array would end up with duplicates of the last item for all of its entries. My theory on why that happens is that the same Java object is being used for the callback and just having its values replaced. It then gets converted to be a Ruby object, but it keeps a reference to the same memory instead of creating a new copy of the object. I think this is the related code:

```
   private IRubyObject yieldResultRows(final ThreadContext context, final Ruby runtime,
            final Connection connection, final ResultSet resultSet,
            final Block block) throws SQLException {

        final ColumnData[] columns = extractColumns(runtime, connection, resultSet, false);

        final IRubyObject[] blockArgs = new IRubyObject[columns.length];
        while ( resultSet.next() ) {
            for ( int i = 0; i < columns.length; i++ ) {
                final ColumnData column = columns[i];
                blockArgs[i] = jdbcToRuby(context, runtime, column.index, column.type, resultSet);
            }
            block.call( context, blockArgs );
        }

        return runtime.getNil(); // yielded result rows
    }
```

Notice that the same object is being used as the arguments object when calling the block. I didn't have a good way of verifying my theory though. I think it could be fixed by creating a new object each time, but I wasn't sure where it was better to fix it. If you think its better to do it on the Java side, then I can try to bite that off in another PR